components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/api/model/ActionDTO.java

```diff
@@ -231,9 +245,9 @@ public static class BuilderForData {
         private final Action.Status status;
         private final EndpointConfig endpoint;
         private final ActionRule rule;
-        private Map<String, Object> properties;
+        private Map<String, ActionPropertyForDAO> properties;
 
-        public BuilderForData(ActionDTO actionDTO) {
+        public BuilderForData(ActionDTO actionDTO, Map<String, ActionPropertyForDAO> properties) {
```

Comment:
why do we bring properties map here ? It's a builder not a constructor of an instance

Reply:
Yeah. Since we already have a setter for properties no need to have the properties as a separate argument to the constructor